tensorrt_llm/_torch/modules/fused_moe/moe_prefetch_manager.py

```diff
@@ -0,0 +1,183 @@
+from typing import List, Tuple
+
+import torch
+
+
+class MoEPrefetchManager:
+    """
+    MoE prefetch manager for MoE weight prefetching.
+    It manages the prefetched layers and buffers.
+    """
+
+    def __init__(self, num_hidden_layers: int, moe_layer_freq: int,
+                 add_one_to_layer_idx: bool, first_k_dense_replace: int,
+                 moe_prefetch_depth: int, moe_prefetch_stride: int):
+        # tp and ep support only
+        self.prefetch_depth = moe_prefetch_depth  # of buffers
+        self.prefetch_stride = moe_prefetch_stride  # prefetch every [stride] moe layers
+        self.weights = {}
+        # set on first register_weight call
+        self.weight_dtype = None
+        self.weight_shapes = None
+        self.device_weight_buffers = []
+
+        # moe layers indices
+        moe_layer_offset = 1 if add_one_to_layer_idx else 0
+        self.moe_layer_indices = [
```

Comment:
Would taking this list as a parameter be a more flexible API. This API seems a little convoluted